share/nisar/examples/noise_spectra_analysis.py

```diff
@@ -0,0 +1,501 @@
+    input_file: str
+        Complete file path for input HDF5 file
+    output_dir: str
+        Output HDF5 file path
+    num_pulses_blk: int
+        Number of pulses in slow time block
+    threshold_margin: float
+        After noise power of raw data is estimated, RFI threshold is derived by
+        adding this threshold margin to it.
+    num_fft: int
+        Number of FFT as well number of samples per segment for the Welch Function.
+        Default: 2048
+    """
+
+    # Output file name is consisted of input file name + SPECTRA.hdf5
+    input_filename = os.path.basename(input_file)
+    input_base, input_ext = os.path.splitext(input_filename)
+    output_file = os.path.join(output_dir, f'{input_base}_SPECTRA{input_ext}')
+    
+    print(f'{output_file = }')
+    print()
+
+    with h5py.File(output_file, 'w') as f_out:
+        # Write metadata as HDF5 attributes
+        f_out.attrs['num_pulses_block'] = num_pulses_blk
+        f_out.attrs['num_fft'] = num_fft
+        f_out.attrs['psd_x_axis'] = 'range (axis=1)'
+        f_out.attrs['psd_y_axis'] = 'power (axis=0, dB)'
+
+        for freq, pol_list in raw.polarizations.items(): # A, B
+            for pol in pol_list: # HH, HV, VV, VH
+                print(f'Start Processing: frequency{freq} {pol}')
+
+                # Read raw data
+                raw_data = raw.getRawDataset(freq, pol)
+                
+                pol_tx = pol[0]
+                fc, fs, _, _ = raw.getChirpParameters(freq, pol_tx)
+
+                # Estimate noise power of each dataset corresponding to a polarization
+                # EVD is used due to unknown RFI situation as a priori
+                noise_pwr_pulse_est_db = estimate_noise_pwr_pulse(raw_data, cpi_len) 
+
+                # Convert Time-Domain Pulse Power to Frequency Domain Threshold in dB/Hz
+                threshold_db_hz = noise_pwr_pulse_est_db - 10*np.log10(fs) + thresh_margin
+
+                num_pulses, _ = raw_data.shape
+                slices = list(slice_gen(num_pulses, num_pulses_blk, combine_rem=False))
+                num_az_blocks = len(slices)
+
+                # Compute Frequencies from dummy data: one pulse
+                freqs, _ = welch(
+                    raw_data[0], 
+                    fs=fs, 
+                    nperseg=num_fft, 
+                    nfft=num_fft, 
+                    detrend=False, 
+                    return_onesided=False
+                )
+
+                # Shift to RF Center Frequency
+                freqs = fftshift(freqs) + fc
+                num_freq_bins = len(freqs)
+
+                # Create output group for output HDF5
+                base_path = f"/science/LSAR/QA/data/freq{freq}/{pol}"
+                out_grp = f_out.require_group(base_path)
+
+                # Center Frequency
+                if "freqCenter" in out_grp:
+                    del out_grp["freqCenter"]
+                dset_fc = out_grp.create_dataset('freqCenter', data=fc)
```

Comment:
```suggestion
                dset_fc = out_grp.require_dataset('centerFrequency', (), np.float64)
                dset_fc[()] = fc
```
Let's spell this the same way as in our other products ([example](https://github-fn.jpl.nasa.gov/NISAR-ADT/NISAR_PIX/blob/83d5e276f7d72093ce571977cec3cba8012508c4/XML/L0/nisar_L0B_RSD.xml#L264)).